lib/html/static/js/fileSearch.js

```diff
@@ -0,0 +1,98 @@
+//function fileLoader(window) {
+function fileLoader(window) {
+
+    // wrapper to highlight the selected line
+    var mark = ['<span class="highlight">', '</span>'];
+    // string used to re-request the file
+
+    var loadFile = function(search_string, goto) {
+        /* Loads in a file from the provided url, populates the line numbers and
+           scrolls to the goto line is provided. */
+
+        // detect search mode
+	search_mode = $("input:radio[name=search_mode]:checked").val();
+	// generate the search url
+        url = getCleanURL() + "&search_string="+search_string+"&search_mode="+
+	    search_mode;
+        $.ajax({
+            url: url,
+            success: function(result) {
+	        result = result.split('---separator---');
+                $('#filecode').html(result[0]);
+                lineNumbers = false;
+		if (search_string != "")
+	            lineNumbers = result[1].match(/[^\r\n]+/g);
+                printLineNumbers(lineNumbers);
+                if (goto)
+	            selectLine(goto);
+            },
+            error: function(error) {
+                alert('Search failed!\nerror opening: '+url);
+            }
+        });
+    }
+
+    var getCleanURL = function() {
+        /* Returns the url of this page with and #x element removed. */
+        return window.location.href.split('#')[0];
+    }
+
+    var selectLine = function(num) {
+        /* Highlights and scrolls to the line of code with the provided
+	 * number. */
+        // insert highlight tags to selected line
+        lines = $('#filecode').html().split('\n');
+        if (num<lines.length) {
+            lines[num-1] = mark[0] + lines[num-1] + mark[1];
+        }
+        $('#filecode').html(lines.join('\n'));
+        // scroll to selected line
+        $('html, body').animate({scrollTop: $('#'+num).offset().top}, 500);
+    }
+
+    var printLineNumbers = function(numbers) {
+        /* Populates #filelinenumbers either with range 1:length(#filecode)
+           or using the provided line numbers. */
+        if (!numbers) { // we are viewing the full file, print all line numbers
+            temp= ''
+            lines = $('#filecode').html().split(/\n/).length;
+            for(i=1; i<lines; i++)
```

Comment:
A general style issue.
* Please add a space between `for` and `(`.
* Please use curly braces for blocks.